serenata_toolbox/federal_senate/federal_senate_dataset.py

```diff
@@ -1,56 +1,95 @@
 import os.path
+from datetime import date
 from urllib.request import urlretrieve
-import numpy as np
+
 import pandas as pd
 
-from datetime import date
 
 class FederalSenateDataset:
     URL = 'http://www.senado.gov.br/transparencia/LAI/verba/{}.csv'
-    FIRST_YEAR = 2008
-    NEXT_YEAR = date.today().year + 1
 
-    YEAR_RANGE = range(FIRST_YEAR, NEXT_YEAR)
+    LAST_YEAR = date.today().year + 1
 
-    def __init__(self, path):
+    def __init__(self, path, first_year=2008, last_year=LAST_YEAR):
         self.path = path
+        self.first_year = first_year
+        self.last_year = last_year
+        self.year_range = range(first_year, last_year)
 
     def fetch(self):
-        for year in self.YEAR_RANGE:
+        retrieved_files = []
+        not_found_files = []
+
+        for year in self.year_range:
             url = self.URL.format(year)
             file_path = os.path.join(self.path, 'federal-senate-{}.csv'.format(year))
-            urlretrieve(url, file_path)
+            try:
+                urlretrieve(url, file_path)
+            except Exception as exception:
```

Comment:
Trying to understand this try/catch during a debug session I'm going through at this moment. What is the reasoning behind catching an exception when one of the files is not found? Shouldn't an exception being raised something wanted in this case?

If that's not the case, at least catching something more specific like `FileNotFoundError` and not `Exception`.